onnxruntime/core/util/math_cpuonly.h

```diff
@@ -20,18 +20,39 @@
 #include "onnxruntime_config.h"
 // external/eigen/Eigen/src/Core/AssignEvaluator.h:86:63:
 // error: enum constant in boolean context [-Werror=int-in-bool-context]
-#if defined(__GNUC__) && __GNUC__ >= 7
+
+#if defined(__GNUC__)
+#pragma GCC diagnostic push
+#if __GNUC__ >= 6
+#pragma GCC diagnostic ignored "-Wignored-attributes"
+#endif
+#pragma GCC diagnostic ignored "-Wunused-parameter"
+#if __GNUC__ >= 7
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wint-in-bool-context"
 #ifdef HAS_DEPRECATED_COPY
 #pragma GCC diagnostic ignored "-Wdeprecated-copy"
 #endif
 #endif
-
+#else
+// build\windows\debug\external\eigen3\unsupported\eigen\cxx11\src/Tensor/Tensor.h(76):
+// warning C4554: '&': check operator precedence for possible error; use parentheses to clarify precedence
```

Comment:
>check operator precedence [](start = 23, length = 25)

I strongly suggest changing code in a way that removes the warning in a simple way, rather than adding these lengthy pragmas.

Reply:
Any solutions ?

Reply:
Isn't the code that generates the warning inside the eigen library? There's no good way to fix that on our side.